src/lib/npm.ts

```diff
@@ -37,45 +56,57 @@ export const ensureNpmDir = async (commandObj: CommandObj): Promise<string> => {
   return npmPath;
 };
 
-type EnsureNpmDependencyOptions = {
-  dependency: Dependency;
+const getVersionToUse = async (
+  recommendedVersion: string,
+  name: NPMDependency,
+  commandObj: CommandObj
+): Promise<string> => {
+  const version = (await initReadonlyDependencyConfig(commandObj)).dependency[
+    name
+  ];
+  return typeof version === "string" ? version : recommendedVersion;
+};
+
+export const npmDependencies: Record<
+  NPMDependency,
+  { recommendedVersion: string; bin: string; packageName: string }
+> = {
+  [AQUA_NPM_DEPENDENCY]: {
+    recommendedVersion: "0.7.4-322",
```

Comment:
So basically I wanted our users to have a predictable experience so I decided to hard-code aqua version that I checked is working correctly with Fluence CLI and also give user this command `fluence dependency` which allows to override recommended version of the dependency

So basically the idea is we allow aqua cli developers to do whatever breaking changes they want and we need to update this version each time we release. I think it's safer to do it like that, but user also has flexibility to switch aqua cli version if he wants to

Reply:
Sounds reasonable. I still think It would be a good idea to extract the version string into some dedicated file, so it would be easier to spot the exact place where dependencies are specified. Once the dependencies are separated we would be able to live with such structure for a while and see if it would be possible to improve.